tasks/install_powershell.ps1

```diff
@@ -107,18 +107,25 @@ function DownloadPuppet {
   Write-Verbose "Downloading the Puppet Agent for Puppet Enterprise on $env:COMPUTERNAME..."
 
   $webclient = New-Object system.net.webclient
-
-  try {
-    $webclient.DownloadFile($msi_source,$msi_dest)
-  }
-  catch [System.Net.WebException] {
-    # If we can't find the msi, then we may not be configured correctly
-    if($_.Exception.Response.StatusCode -eq [system.net.httpstatuscode]::NotFound) {
-        Throw "Failed to download the Puppet Agent installer: $msi_source"
+  
+  $a = 0
+  do {
+    try {
+      $webclient.DownloadFile($msi_source,$msi_dest)
+      break
     }
-    # Throw all other WebExceptions
-    Throw $_
-  }
+    catch [System.Net.WebException] {
+      # If we can't find the msi, then we may not be configured correctly
+      if($_.Exception.Response.StatusCode -eq [system.net.httpstatuscode]::NotFound) {
+          Throw "Failed to download the Puppet Agent installer: $msi_source"
+      }
+      # Throw all other WebExceptions
+      if($a -eq $retry) {
+        Throw $_
+      }
+    }
+    $a++
+  } while($a -le $retry)
```

Comment:
Would `$a` ever be equal to `$retry` and throw the exception?

Reply:
I should because the while condition is `$a -le $retry`(less or equal). So when $a hits the same value as $retry, the loop will still iterate once again.